x-pack/plugins/security_solution/server/lib/detection_engine/rule_actions/deletion_migration/update_objects.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UpdateObjectsOptions } from '../../../startup_migrations/types';
+
+import { SavedObject, SavedObjectsBulkUpdateObject } from '../../../../../../../../src/core/server';
+import { Alerting, SideCarAction } from '../deletion_migration/types';
+import { apiKeyAsAlertAttributes } from './api_key_as_alert_attributes';
+import { getActionsCountInReferences } from '../deletion_migration/get_actions_count_in_references';
+import { transformActions } from '../deletion_migration/transform_actions';
+import { getThrottle } from './get_throttle';
+
+export const updateObjects = async ({
+  savedObject: actionSideCar,
+  joins,
+  encryptedSavedObjectsClient,
+  logger,
+}: UpdateObjectsOptions<SideCarAction>): Promise<Array<SavedObjectsBulkUpdateObject<Alerting>>> => {
+  logger.debug(
+    `Attempting to update alerting objects to remove the legacy action of type, siem-detection-engine-rule-actions and id of "${actionSideCar.id}"`
+  );
+
+  if (encryptedSavedObjectsClient == null) {
+    logger.error(
+      [
+        'The "encryptedSavedObjects" plugin was not found within the security_solution application.',
+        'Could not update legacy actions to newer actions because this plugin is not found',
+        'Please check the existing "security_solution" actions to ensure their action intervals are set correctly as migrations have failed',
+      ].join(' ')
+    );
+    return [];
+  }
+
+  const [resolvedAlert, ...resolvedActions] = joins as [
+    SavedObject<Alerting>,
+    ...Array<SavedObject<unknown>>
+  ];
+
+  logger.debug(`Updating alert id: "${resolvedAlert.id}" to remove legacy actions`);
+
+  const keyAttributes = await apiKeyAsAlertAttributes({
+    logger,
+    id: resolvedAlert.id,
+    encryptedSavedObjectsClient,
+  });
+
+  if (keyAttributes == null) {
+    logger.error(
+      [
+        'Key attributes were found to be "null" or "undefined", returning without updating.',
+        'Please check the existing "security_solution" actions to ensure their action intervals are set correctly as migrations have failed',
+      ].join(' ')
+    );
+    return [];
+  }
+
+  if (actionSideCar.attributes.actions.length !== resolvedActions.length) {
```

Comment:
From @yctercero during in-person review with @dhurley14 these might not be 1-1. So I need to remove this check here and write tests (if this goes off of draft mode)

Reply:
Actually I think these will be 1-1 and they will just have duplicates. I think we are ok here. More testing will be utilized for this point.